### PR TITLE
Nightly Reports: Generate markdown slack report from new nightlies

### LINKED ISSUE #25435: Update Connector Health Reports to use new Dagger test statuses
## Overview
We want to update our slack connector health reports  generated currently from [report.sh](https://github.com/airbytehq/airbyte/blob/master/tools/status/report.sh#L106) to use the new dagger test reports.

This may or may not involve updating the system to use the new orchestrator to monitor for test report changes

### PR DESCRIPTION
**Pull request by bnchrch**
## Goal
On new nightly report notify slack of outcome

## What was done
image

1. Refactored a gcs sensor to be generic
2. Added a new github action sensor (which isnt being used but it does work)
3. Added a slack webhook op
4. Added a new nightly sensor
5. Added 2 new gcs resources for the nightly outputs
6. Added a new job that generates the report and posts it to slack

## Reading order
1. [.../connectors/metadata_service/orchestrator/orchestrator/assets/connector_nightly_report.py](https://github.com/airbytehq/airbyte/pull/26990/files#diff-50ecf787aaf128040b91f0658c6fe267a2121f3b9e1f740f983e1bcc303816d1)
2. [airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/templates/render.py](https://github.com/airbytehq/airbyte/pull/26990/files#diff-a55f8aa9ae28537edc341d46ce9fe82f37f578b1df44b54354132526f0ec25fb)
3. [airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/ops/slack.py](https://github.com/airbytehq/airbyte/pull/26990/files#diff-98abe5a1dcb80ee786e7156d053e4dada2d61d1e10a399c0f670fee2359aa3d5)
4. [airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/sensors/gcs.py](https://github.com/airbytehq/airbyte/pull/26990/files#diff-3ea054ad13c9cd12ddaa252c0e30fe940977c81e5dabc5bac9318da380a0637a)

## Before merge

- [x] Set `NIGHTLY_REPORT_SLACK_WEBHOOK_URL` on prod dagster
- [x] Ensure metadata sa has airbyte-ci-report access

closes #25435

### CODE REVIEW

File: airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/assets/connector_nightly_report.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    return models_df
+
+
+def get_latest_reports(blobs, number_to_get):
+    # We can sort by the name to get the latest 10 nightly runs
+    latest_nightly_complete_file_blobs = sorted(blobs, key=lambda blob: blob.name, reverse=True)[:number_to_get]
+    return latest_nightly_complete_file_blobs
+
+
+def get_relevant_test_outputs(latest_nightly_test_output_file_blobs, latest_nightly_complete_file_blobs):
```
Comment (alafanechere) [inline airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/assets/connector_nightly_report.py:48]: Is this the function retrieving connector specific test reports? 
I'd appreciate a docstring to explain the logic 😄

File: airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/models/ci_report.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    failed_steps: Optional[list] = None
+    successful_steps: Optional[list] = None
+    skipped_steps: Optional[list] = None
+    gha_workflow_run_url: Optional[str] = None
+    pipeline_start_timestamp: Optional[int] = None
+    pipeline_end_timestamp: Optional[int] = None
+    pipeline_duration: Optional[int] = None
+    git_branch: Optional[str] = None
+    git_revision: Optional[str] = None
+    ci_context: Optional[str] = None
```
Comment (alafanechere) [inline airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/models/ci_report.py:21]: nit: Would be cool to import the types from the `pipelines` python package and deserialize to the CIContext enum.

File: airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/ops/slack.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+            msg = ""
+    yield msg
+
+
+@op
+def send_slack_webhook(webhook_url, report):
+    webhook = WebhookClient(webhook_url)
+    for msg in chunk_messages(report):
+        # Wrap in code block as slack does not support markdown in webhooks
+        webhook.send(text=f"```{msg}```")
```
Comment (alafanechere) [inline airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/ops/slack.py:21]: nit: It'd be great to add the total duration of the nightly build pipeline, maybe something the github api can provide with the run url. We can do it later.

File: airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/templates/render.py [hunk 4]
```diff
@@ -72,10 +72,113 @@ def dataframe_to_table_html(df: pd.DataFrame, column_mapping: List[ColumnInfo])
...
 
+def value_to_emoji(value: Any) -> str:
+    if value is True:
+        return "✅"
+    elif value is False:
+        return "❌"
+    elif value is None or pd.isna(value):
+        return "❓"
+    else:
+        return str(value)
```
Comment (alafanechere) [inline airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/templates/render.py:83]: nit: use a mapping:
```python
value_to_emoji = {True: "✅", False: ,"❌", None: "❓"}
return value_to_emoji.get(value, str(value))
```

**Review by alafanechere — APPROVED** [review]
👍 I added a couple of comments just to get clearer explanation abouth the connector report retrieval logic. It's not blocking as the report sent to #publish-on-merge updates look good. Thanks!

File: airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/models/ci_report.py (hunk 2 above)
Comment (bnchrch) [reply airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/models/ci_report.py:21]: Agreed! Ill add a TODO here as that will be easier to do once pipelines is in the airbyte-ci folder

File: airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/templates/render.py (hunk 4 above)
Comment (bnchrch) [reply airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/templates/render.py:83]: I like this! but unfortunately we cant use Bools as keys 😢 , We could stringify the value before getting but then the None case gets strange.

**Pull request merged**